sdk/spring/azure-spring-cloud-context/src/main/java/com/azure/spring/cloud/autoconfigure/context/AzureContextAutoConfiguration.java

```diff
@@ -72,4 +74,17 @@ public SubscriptionSupplier subscriptionSupplier(AzureResourceManager azureResou
         return azureResourceManager::subscriptionId;
     }
 
+    @Bean
+    @ConditionalOnMissingBean
+    @ConditionalOnBean(AzureResourceManager.class)
+    public ResourceGroupManager resourceGroupManager(AzureResourceManager azureResourceManager,
+                                                         AzureProperties azureProperties) {
+        ResourceGroupManager resourceGroupManager = new ResourceGroupManager(azureResourceManager, azureProperties);
+        if (azureProperties.isAutoCreateResources()
```

Comment:
I'm not sure if this is the right place to do this creation , since this will slow the spring boot app startup time

Reply:
Yes, it will slow the app startup. The resource group is the prerequisite for other resources, and the creation of other Beans will depend on an existing resource group, for example: eventhubs kafkaproperties will [getOrCreate the eventhub ](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/spring/azure-spring-cloud-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/eventhub/AzureEventHubKafkaAutoConfiguration.java#L65)namespace, these all happened after the automatic resource creation switch was turned on.

Reply:
then probably add some debug level log to state this case

Reply:
Added.